src/vivarium_public_health/plugins/parser.py

```diff
@@ -334,3 +333,407 @@ def get_data_getter(
+                        error_messages.append(
+                            f"External configuration for package '{package}' must "
+                            "be a list of paths to yaml files."
+                        )
+        if error_messages:
+            raise ConfigurationError("\n".join(error_messages), None)
+
+    def validate_causes_config(self, causes_config: ConfigTree) -> None:
+        """
+        Validates the cause model configuration.
+
+        Parameters
+        ----------
+        causes_config
+            A ConfigTree defining the cause model configurations
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ConfigurationError
+            If the cause model configuration is invalid
+        """
+        causes_config = causes_config.to_dict()
+        error_messages = []
+        for cause_name, cause_config in causes_config.items():
+            error_messages += self._validate_cause(cause_name, cause_config)
+
+        if error_messages:
+            raise ConfigurationError("\n".join(error_messages), None)
+
+    def _validate_cause(self, cause_name: str, cause_config: Dict[str, Any]) -> List[str]:
+        """
+        Validates a cause configuration and returns a list of error messages.
+
+        Parameters
+        ----------
+        cause_name
+            The name of the cause to validate
+        cause_config
+            A ConfigTree defining the cause to validate
+
+        Returns
+        -------
+        List[str]
+            A list of error messages
+        """
+        error_messages = []
+        if not isinstance(cause_config, dict):
+            error_messages.append(
+                f"Cause configuration for cause '{cause_name}' must be a dictionary."
+            )
+            return error_messages
+
+        if not set(cause_config.keys()).issubset(
+            CausesConfigurationParser.ALLOWABLE_CAUSE_KEYS
+        ):
+            error_messages.append(
+                f"Cause configuration for cause '{cause_name}' may only"
+                " contain the following keys: "
+                f"{CausesConfigurationParser.ALLOWABLE_CAUSE_KEYS}."
+            )
+
+        model_type = cause_config.get("model_type", "")
+        if model_type and not (isinstance(model_type, str) and "." in model_type):
+            error_messages.append(
+                f"If 'model_type' is provided for cause '{cause_name}' it "
+                "must be a fully qualified import path for the type of the "
+                f"desired model. Provided'{model_type}'."
+            )
+
+        states_config = cause_config.get("states", {})
+        if not states_config:
+            error_messages.append(
+                f"Cause configuration for cause '{cause_name}' must define "
+                "at least one state."
+            )
+
+        if not isinstance(states_config, dict):
```

Comment:
Wouldn't this mean cause_config is just incorrectly defined?  i.e. why/where can we use a states_config that is not a dict?

Reply:
Exactly, if `states_config` is not a dict the configuration is invalid, so we'll throw and error.